src/NzbDrone.Common/Disk/DiskTransferService.cs

```diff
@@ -83,6 +84,100 @@ public TransferMode TransferFolder(string sourcePath, string targetPath, Transfe
             return result;
         }
 
+        public int MirrorFolder(string sourcePath, string targetPath)
+        {
+            var filesCopied = 0;
+
+            Ensure.That(sourcePath, () => sourcePath).IsValidPath();
+            Ensure.That(targetPath, () => targetPath).IsValidPath();
+
+            _logger.Debug("Mirror [{0}] > [{1}]", sourcePath, targetPath);
+
+            if (!_diskProvider.FolderExists(targetPath))
+            {
+                _diskProvider.CreateFolder(targetPath);
+            }
+
+            var sourceFolders = _diskProvider.GetDirectoryInfos(sourcePath);
+            var targetFolders = _diskProvider.GetDirectoryInfos(targetPath);
+
+            foreach (var subDir in targetFolders.Where(v => !sourceFolders.Any(d => d.Name == v.Name)))
+            {
+                _diskProvider.DeleteFolder(subDir.FullName, true);
+            }
+
+            foreach (var subDir in sourceFolders)
+            {
+                filesCopied += MirrorFolder(subDir.FullName, Path.Combine(targetPath, subDir.Name));
+            }
+
+            var sourceFiles = _diskProvider.GetFileInfos(sourcePath);
+            var targetFiles = _diskProvider.GetFileInfos(targetPath);
+
+            foreach (var targetFile in targetFiles.Where(v => !sourceFiles.Any(d => d.Name == v.Name)))
+            {
+                _diskProvider.DeleteFile(targetFile.FullName);
+            }
+
+            foreach (var sourceFile in sourceFiles)
+            {
+                var targetFile = Path.Combine(targetPath, sourceFile.Name);
+
+                if (CompareFiles(sourceFile.FullName, targetFile))
+                {
+                    continue;
+                }
+
+                TransferFile(sourceFile.FullName, targetFile, TransferMode.Copy, true, true);
+                filesCopied++;
+            }
+
+            return filesCopied;
+        }
+
+        private bool CompareFiles(string sourceFile, string targetFile)
+        {
+            if (_diskProvider.FileExists(sourceFile) || _diskProvider.FileExists(targetFile))
```

Comment:
I wasn't concerned about the check, but the logic, if either file exists we return false which means we never actually check to see if they are the same before copying the file.

You need two checks here:

```
if (!_diskProvider.FileExists(sourceFile))
{
    return true;
}
```

So it doesn't try to transfer the file.

```
if (!_diskProvider.FileExists(targetFile))
{
    return false;
}
```

So the source file is copied because the target file doesn't exist.